### DISCUSSION TITLE
Elements not loading through CloudFlare tunnel

### DISCUSSION THREAD

**Discussion by petri-ojala** [mcguirepr89/BirdNET-Pi#383] [General]
I configured BirdNet-Pi behind Cloudflare's tunnel.  Super easy to configure with cloudflared, highly recommended.  Opening inbound NAT right at my home network sounds a bit too 90's.

Feel free to access at https://jokipuisto-birdnet.ojala.cloud/

Some of the elements don't seem to load though, they work fine locally.  Especially on the Overview page.

It seems to be calling overview.php every second or so, one of these failed with 500 (also visible in Caddy logs).  Looks a bit heavy and not sure what on the Overview page needs such a frequently loading.
Any thoughts what is causing this?

**Comment by ehpersonal38** [comment]
Funnily enough I think I fixed this issue yesterday - could you try an update by going to Tools -> System Controls -> Update, and see if this issue persists?

**Reply by petri-ojala** [reply]
Updated to the latest version (also restarted all services) but doesn't seem to help.

**Reply by ehpersonal38** [reply]
Sorry to hear that. Given we had the [exact same issue](https://github.com/mcguirepr89/BirdNET-Pi/issues/380) yesterday that was resolved by a commit in the latest version, I can only assume that your system is still using an old version of the script (that version made an unnecessary HTTP call to the BirdNET-Pi web server, which has problems behind reverse proxies). So before I go further debugging, could you please paste the output of this command:

`cat $HOME/BirdNET-Pi/scripts/overview.php | sed -n '47,55p'`

**Reply by petri-ojala** [reply]
```
birdnet:~/BirdNET-Pi$ cat $HOME/BirdNET-Pi/scripts/overview.php | sed -n '47,55p'
  while($mostrecent = $result4->fetchArray(SQLITE3_ASSOC)) {
    $comname = preg_replace('/ /', '_', $mostrecent['Com_Name']);
    $sciname = preg_replace('/ /', '_', $mostrecent['Sci_Name']);
    $comname = preg_replace('/\'/', '', $comname);
    $filename = "/By_Date/".$mostrecent['Date']."/".$comname."/".$mostrecent['File_Name'];

      // check to make sure the image actually exists, sometimes it takes a minute to be created
      if (isset($_SERVER['HTTPS']) &&
          ($_SERVER['HTTPS'] == 'on' || $_SERVER['HTTPS'] == 1) ||
```

**Reply by petri-ojala** [reply]
Just FYI, I have a real DNS name also configured for it locally (http://birdnet.ojala.cloud) and it's working fine.  mDNS doesn't work with all the devices.

**Reply by ehpersonal38** [reply]
Yeah so as I suspected that's still the old script. Something must be wrong with your update process. Could you do an update again and paste the output here?

**Reply by petri-ojala** [reply]
Ok, fixed now and working.  For some reason `.git/objects` had directories with root as the owner, not birdnet and it did complain about it but the update script went through and the icon about updates waiting disappeared.  
Not sure when/how it has happened, I've been staying away from doing anything manually or touching the files as I broke the initial installation after a week, tried to fix some issues by modifying the scripts and ending up in a state that I couldn't keep up with the official version.  I have only updated it through the button since then every now and then.

Good to see that more people are working on the code, one day I'll find enough time to figure out the logic and can contribute as well :-)   I'd like to turn the RasPi part just to a container that does the recordings, and move all the processing, frontend and backend to containers running on my home Kubernetes cluster with Longhorn storage.   (I'm using the same RasPi for ADS-B receiver, running containers to feed the data to various services..)

**Reply by ehpersonal38** [reply]
Glad to hear you solved it! Unfortunately I caused this one - I had a PHP script call `sudo git` for a while before Patrick came along and cleaned up my mess. 

Also there was some work being done with docker here, you might be interested: https://github.com/mcguirepr89/BirdNET-Pi/issues/211

**Reply by petri-ojala** [reply]
No worries :-)   Thanks for all the help.   Good to see that Oliver has been active on the containerization.

**Comment by Svardsten53** [comment]
Moi Petri! I had exactly the same issue as you and the fix by **ehpersonal38** yesterday solved it. I hope you also get it solved.
https://github.com/mcguirepr89/BirdNET-Pi/issues/380

**Comment by ehpersonal38** [comment]
If anyone else reading this has this issue, follow [this](https://github.com/mcguirepr89/BirdNET-Pi/issues/355#issuecomment-1150966226) to fix it!
